main.tf

```diff
@@ -195,7 +195,7 @@ module "node_pools" {
   max_pods                     = each.value.max_pods == null ? 110 : each.value.max_pods
   node_taints                  = each.value.node_taints
   node_labels                  = each.value.node_labels
-  zones                        = (var.node_pools_availability_zone == "" || var.node_pools_proximity_placement == true) ? [] : [var.node_pools_availability_zone]
+  zones                        = (var.node_pools_availability_zone == "" || var.node_pools_proximity_placement == true) ? [] : (var.node_pools_availability_zone != "1") ? [var.node_pools_availability_zone] : var.node_pools_availability_zones
```

Comment:
Why not just have the logic like this:

```yaml
zones = (var.node_pools_availability_zone == "" || var.node_pools_proximity_placement == true) ? [] :
  length(var.node_pools_availability_zones) != 0 ?
    var.node_pools_availability_zones : [var.node_pools_availability_zone]
```
With this logic we can accurately set the default value of the `node_pools_availability_zones` to `null` and fix the logic when we deprecate the older variable.

Seems like there is a way to clean up this logic unless the value of the first zone in a location is always needed, then the logic can stay.

Reply:
Verified by running test with `node_pools_availability_zones` defaulted to `null` and `node_pools_availability_zone` to empty string. The first zone value does not seem to be required. Setting the empty list of string for Zones just sets the `Availability zones` to `None` for the nodes.
I'll update the logic using your suggested condition.